src/utils.ts

```diff
@@ -1,15 +1,54 @@
 import { SelectionEntry, UIDisplayData } from '@aics/simularium-viewer';
 import { VisibilitySelectionMap } from './constants';
 
-export const getNewSelectionMap = (
+export const getNewMapAfterTopLevelCheckboxClick = (
   agentName: string,
   currentVisibilityMap: VisibilitySelectionMap
 ): VisibilitySelectionMap => {
   const newMap: VisibilitySelectionMap = { ...currentVisibilityMap };
-  if (currentVisibilityMap[agentName].length === 0) {
+  if (currentVisibilityMap[agentName].includes(agentName)) {
+    newMap[agentName] = [];
+  } else {
     newMap[agentName] = [agentName];
+  }
+  return newMap;
+};
+
+export const getNewMapAfterDisplayStateClick = (
```

Comment:
The name DisplayState doesn't really suggest the parent/child relationship between agents and display states. Is there a better name that would make this more clear?

Reply:
Could we just call them `agentChildDisplayState`?

Reply:
I agree that `displayState` name is clunky, but that typing is defined in the viewer so I think it's out of scope for this PR, but it's definitely one of the things about the `UiDisplayData` data structure that could be improved.

My reasoning behind sticking with that naming convention in the front end, where we could just substitute 'child' here in the function name, (`getNewMapAfterChildCheckboxClick` for example) was to keep explicit what data structures we are using to arrive at these components, because there is a lot of converting back and forth and use of different terminology that makes harder to follow what is derived from what (hide, highlight, visibility, selection or: agent, displayState, topLevel, shared, child, etc.)